### ISSUE TITLE
Implement “Zombie Stall Ping” to know if stalls are alive

### ISSUE THREAD

**Issue by ktecho** [PlebeianApp/plebeian.market#429]
One of the biggest problems of nip-15 marketplaces is spending the time to look through products, read descriptions, make the buying decission, only to discover in the last step that the marketplace is not active. Maybe the computer in which it runs is off, or it's an abandoned stall.

Having a way to ping stalls would be awesome, so maybe it should be proposed for the nip-15. But in the mainwhile, it can be implemented in a quick-and-dirty way. It would be so easy as creating a "checkout event" (step 1) with a non-existing product ID. The marketplace will reply with an error, that you'll interpret as "marketplace alive".

It can be done automatically when you're visiting each stall or each product (with local cache so we don't bother the marketplace all the time. If it was alive 15 minutes ago, it's probably alive now).

**Comment by gzuuus** [comment]
I'm not sure if I understand your point. You seem to be mixing up marketplaces and stalls. Marketplaces are instances that someone can run, whereas stalls are individuals creating Nostr events that can be accessed through Nostr, and marketplaces. The key point is that marketplaces and stalls are decoupled, meaning that stalls don't require a specific marketplace to operate.
If I understand your proposal correctly, you're suggesting two separate mechanisms: one to detect whether marketplaces are online, and another to check whether stalls are active. Detecting whether marketplaces are online is relatively straightforward, as it can be done by pinging them or making an API call.
However, verifying whether stalls are active is more complex, as it requires interacting with the individual behind the stall to confirm their presence. Currently, PM allows users to place an order and pay if the merchant has set up payment details, or simply place an order if the merchant doesn't have payment details or if the user wants to check if someone is behind the stall before making a payment

**Comment by ktecho** [comment]
What I want to avoid is knowing that a stall is not going to work at the last step because there is nobody listening for the Nostr events. It can be because the stall has been abandoned or because the server/node has temporary problems.

**Comment by gzuuus** [comment]
Yes, at that point, you can simply place the order by sending the proper status messages and wait for a response. A stall itself is just a Nostr event. The utility layer is the client that enables you to interact with the stall and communicate with the owner, correct?

**Comment by ktecho** [comment]
Yes, but maybe an owner doesn't exist anymore because it was a test stall, or the seller was not selling anything and just shut-down the app... It would be cool to know in advance if the stall is active.

**Comment by gzuuus** [comment]
Yes, eventually we can use different metrics we can collect to determine last activity and so on

**Comment by yojimble** [comment]
yes please - anything but an "is this available" message box!!!!!!!!!
